### ISSUE TITLE
Request to not immediately apply log mode changes in [nbx]/sliders

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#1648]
If setting to log mode while min/max touches or crosses zero, the range values are immediately set. Maybe you accidentally clicked on it or changed your mind or something then you need to set the values back to what they were.

Seems best to apply this change only when you hit ok. What do you think?

Also, I am proposing a new mode and it would be better to navigate between the options this way -- see https://github.com/pure-data/pure-data/issues/1647

**Comment by umlaeute** [comment]
it doesn't apply *immediately* on Linux/Windows (you have to click <kbd>Apply</kbd> resp <kbd>OK</kbd> first).
afaiu, "immediate apply" is a GUI guideline by Apple and expected by macOS users (and there has been plenty of discussion)

so i think a better idea is to not disallow changing from lin to log mode, if we cannot do a bidrectional mapping (going back from log to lin without losing anything).

**Comment by Ant1r** [comment]
> to not disallow

"to disallow" I guess;-)

**Comment by umlaeute** [comment]
yes.
fixing my comment...

**Comment by porres** [comment]
>  i think a better idea is to disallow changing from lin to log mode, if we cannot do a bidrectional mapping (going back from log to lin without losing anything).

That would mean not being able to set this at all in properties, right? I think it's best to just live with this instead and close this issue.

**Comment by porres** [comment]
yeah, just closing this, unless I got it wrong, so tell me and i'll reopen

**Comment by porres** [comment]
Ok, I thought of a way to go back and forth the log/linear settings. It's very simple, just do not apply any made up new configuration, let it be wrong. If it is wrong in the end when changes were applied, just don't output any number and give an error instead, something like "range cannot contain '0' in log mode". If this is automatically applied in the mac, then the error will be there when doing changes, which is fine. What do you think?
